ddl/multi_schema_change.go

```diff
@@ -56,18 +57,135 @@ func (d *ddl) MultiSchemaChange(ctx sessionctx.Context, ti ast.Ident) error {
 	return d.callHookOnChanged(job, err)
 }
 
-func checkMultiSchemaInfo(info *model.MultiSchemaInfo, t table.Table) error {
-	err := checkOperateSameColAndIdx(info)
-	if err != nil {
-		return err
+func onMultiSchemaChange(w *worker, d *ddlCtx, t *meta.Meta, job *model.Job) (ver int64, err error) {
+	if job.MultiSchemaInfo.Revertible {
+		// Handle the rolling back job.
+		if job.IsRollingback() {
+			// Rollback/cancel the sub-jobs in reverse order.
+			for i := len(job.MultiSchemaInfo.SubJobs) - 1; i >= 0; i-- {
+				sub := job.MultiSchemaInfo.SubJobs[i]
+				if sub.IsFinished() {
+					continue
+				}
+				proxyJob := sub.ToProxyJob(job)
+				ver, err = w.runDDLJob(d, t, proxyJob)
+				sub.FromProxyJob(proxyJob)
+				if i == 0 && sub.IsFinished() {
+					job.State = model.JobStateRollbackDone
+				}
```

Comment:
Could we remove this logic? It seems that we will do it on line79.

Reply:
This is a tiny optimization so that we don't have to wait one more round.